### DISCUSSION TITLE
How will Reticulum work with multiple nodes based on lora?

### DISCUSSION THREAD

**Discussion by Eninspace** [markqvist/Reticulum#165] [General]
How will Reticulum work with multiple nodes based on lora? If node A sends a message to node B via multiple lora hops (without direct visibility), will other lora-based nodes orginized and know routes? I mean does Reticulum allow to organize a mesh network at the transport level of lora?

**Comment by markqvist** [comment]
Yes, of course, it will work fine :)

Reticulum allows creating self-organising mesh networks out of any of the medium types it supports. This can be of a single type or a mix of any number of types.

For slow mediums such as LoRa, the most efficient methodology is to set up the network in a way such that nodes will not have to directly repeat packets on their slow interfaces (LoRa or packet radio for example). But if that is all you have to work with, it *will* work, it will just add more latency per hop and lower throughput.

If you can for example have separate uplink and dowlink channels (for example two RNodes) on each transport instance (Reticulum nodes that do routing), the nodes can send on packets at the same time it is receiving them (full duplex operation). If you only have one LoRa radio on each transport instance, they will have to run the routing in half-duplex operation, ie., receive a packet, tie up the channel while sending it on to the next hop, while the initial sender is now waiting for the channel to become clear again. When the initial sender notices the channel is clear again, it can now send the next packet, and so on. As such, you can see it is slower method, but it does work.
